Mage.Sets/src/mage/cards/s/SunbirdEffigy.java

```diff
@@ -0,0 +1,194 @@
+import mage.abilities.keyword.HasteAbility;
+import mage.abilities.keyword.VigilanceAbility;
+import mage.abilities.mana.SimpleManaAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.SubType;
+import mage.constants.Zone;
+import mage.game.ExileZone;
+import mage.game.Game;
+import mage.util.CardUtil;
+
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+/**
+ * @author TheElk801
+ */
+public final class SunbirdEffigy extends CardImpl {
+
+    public SunbirdEffigy(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT, CardType.CREATURE}, "");
+
+        this.subtype.add(SubType.BIRD);
+        this.subtype.add(SubType.CONSTRUCT);
+        this.power = new MageInt(0);
+        this.toughness = new MageInt(0);
+        this.nightCard = true;
+
+        // Flying
+        this.addAbility(FlyingAbility.getInstance());
+
+        // Vigilance
+        this.addAbility(VigilanceAbility.getInstance());
+
+        // Haste
+        this.addAbility(HasteAbility.getInstance());
+
+        // Sunbird Effigy's power and toughness are each equal to the number of colors among the exiled cards used to craft it.
+        this.addAbility(new SimpleStaticAbility(
+                Zone.ALL, new SetBasePowerToughnessSourceEffect(SunbirdEffigyValue.instance)
+        ).addHint(SunbirdEffigyHint.instance));
+
+        // {T}: For each color among the exiled cards used to craft Sunbird Effigy, add one mana of that color.
+        this.addAbility(new SimpleManaAbility(Zone.BATTLEFIELD, new SunbirdEffigyEffect(), new TapSourceCost()));
+    }
+
+    private SunbirdEffigy(final SunbirdEffigy card) {
+        super(card);
+    }
+
+    @Override
+    public SunbirdEffigy copy() {
+        return new SunbirdEffigy(this);
+    }
+}
+
+enum SunbirdEffigyValue implements DynamicValue {
+    instance;
+
+    @Override
+    public int calculate(Game game, Ability sourceAbility, Effect effect) {
+        ExileZone exileZone = game
+                .getExile()
+                .getExileZone(CardUtil.getExileZoneId(
+                        game,
+                        sourceAbility.getSourceId(),
+                        game.getState().getZoneChangeCounter(sourceAbility.getSourceId()) - 2
+                ));
+        return exileZone == null ? 0 : exileZone
+                .getCards(game)
+                .stream()
+                .map(card -> card.getColor(game))
+                .reduce(new ObjectColor(), (c1, c2) -> c1.union(c2))
+                .getColorCount();
+    }
+
+    @Override
+    public SunbirdEffigyValue copy() {
+        return this;
+    }
+
+    @Override
+    public String getMessage() {
+        return "colors among the exiled cards used to craft it";
+    }
+
+    @Override
+    public String toString() {
+        return "1";
+    }
+}
+
+enum SunbirdEffigyHint implements Hint {
```

Comment:
(I wish hints could leverage more of the dynamic value code rather than duplicating code, but that's a whole separate thing)

Reply:
I can actually de-dupe some of the code here